src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.model.User;
+import java.util.concurrent.atomic.AtomicLong;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static RegistrationServiceImpl registrationService;
+    private static AtomicLong userIdCounter;
+
+    @BeforeAll
+    static void createReference() {
+        registrationService = new RegistrationServiceImpl();
+        userIdCounter = new AtomicLong(1);
+    }
```

Comment:
Consider whether you should use `@BeforeAll` or `@BeforeEach` for initializing `registrationService`. `@BeforeAll` is used for static initializations that are done once before all tests, while `@BeforeEach` is used for initializations that should happen before each test. If `registrationService` does not maintain any state between tests, `@BeforeEach` might be more appropriate to ensure a clean state for each test.